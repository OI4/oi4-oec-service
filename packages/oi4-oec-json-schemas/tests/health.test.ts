import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/health.schema.json'
import deviceHealthEnumerationSchema from '../src/schemas/constants/DeviceHealthEnumeration.schema.json'

import validHealthObjs from './__fixtures__/healths_valid.json'
import invalidHealthObjs from './__fixtures__/healths_invalid.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  }, (ajv) => {
    ajv.addSchema(deviceHealthEnumerationSchema)
  })
)

describe('health schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it.each(validHealthObjs as [])(
    '(%#) match valid config object to schema -> %s',
    (_name: string, obj) => {
      expect(obj).toMatchSchema(schema)
    }
  )

  it.each(invalidHealthObjs)(
    '(%#) match fails for invalid config -> %s',
    (name: string, obj) => {
      expect(obj).not.toMatchSchema(schema)
      expect(() => {
        expect(obj).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(name)
    }
  )
})
