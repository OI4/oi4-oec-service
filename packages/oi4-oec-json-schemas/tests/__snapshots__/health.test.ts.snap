// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`health schema (0) match fails for invalid config -> empty: empty 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'health'[39m
[31m    Path:     health.schema.json#/required[39m
[31m[39m"
`;

exports[`health schema (1) match fails for invalid config -> health is missing: health is missing 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'health'[39m
[31m    Path:     health.schema.json#/required[39m
[31m[39m"
`;

exports[`health schema (2) match fails for invalid config -> health is not within enum: health is not within enum 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /health must be equal to one of the allowed values[39m
[31m    Expected: [\\"NORMAL_0\\",\\"FAILURE_1\\",\\"CHECK_FUNCTION_2\\",\\"OFF_SPEC_3\\",\\"MAINTENANCE_REQUIRED_4\\"][39m
[31m    Received: [33m<string>[39m[31m blub[39m
[31m    Path:     health.schema.jsonconstants/DeviceHealthEnumeration.schema.json/enum[39m
[31m[39m"
`;

exports[`health schema (3) match fails for invalid config -> health score is larger 100: health score is larger 100 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /healthScore must be <= 100[39m
[31m    Received: [33m<number>[39m[31m 101[39m
[31m    Path:     health.schema.json#/properties/healthScore/maximum[39m
[31m[39m"
`;

exports[`health schema (4) match fails for invalid config -> health score is negative: health score is negative 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /healthScore must be >= 0[39m
[31m    Received: [33m<number>[39m[31m -1[39m
[31m    Path:     health.schema.json#/properties/healthScore/minimum[39m
[31m[39m"
`;

exports[`health schema (5) match fails for invalid config -> swapped key and value: swapped key and value 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /health must be string[39m
[31m    Received: [33m<number>[39m[31m 0[39m
[31m    Path:     health.schema.jsonconstants/DeviceHealthEnumeration.schema.json/type[39m
[31m[39m[31m  /health must be equal to one of the allowed values[39m
[31m    Expected: [\\"NORMAL_0\\",\\"FAILURE_1\\",\\"CHECK_FUNCTION_2\\",\\"OFF_SPEC_3\\",\\"MAINTENANCE_REQUIRED_4\\"][39m
[31m    Received: [33m<number>[39m[31m 0[39m
[31m    Path:     health.schema.jsonconstants/DeviceHealthEnumeration.schema.json/enum[39m
[31m[39m[31m  /healthScore must be integer[39m
[31m    Received: [33m<string>[39m[31m CHECK_FUNCTION_2[39m
[31m    Path:     health.schema.json#/properties/healthScore/type[39m
[31m[39m"
`;
